### ISSUE TITLE
rename tsx files doesn't change them on disk

### ISSUE THREAD

**Issue by kushalmahajan** [Microsoft/vscode#67926]
renaming tsx files through vscode reflects in file explorer but doesn't changes on disk. 

MacOS - 10.13.6
VsCode - 1.30.2

**Comment by bpasero** [comment]
@kushalmahajan what exactly are you doing and what is the result? any errors in the log?

**Comment by kushalmahajan** [comment]
Why exactly I'm doing?

I converted all my project files to lowercase.

Result is that case sensitive plugin in CRA detects that and makes me aware
of myth.

I cross checked it on bash and renamed it from there to have the project
running.

So probably, it's vscode. Right?

Warm Regards,
Kushal V. Mahajan

**Comment by bpasero** [comment]
@kushalmahajan what about a normal text file, when you rename that, is that reflected on the file system?

**Comment by kushalmahajan** [comment]
Normal creation is fine. I think issue arises when I push on git with uppercase. 

Here's the situation.  My other colleague is working on windows. 

After I try to change the uppercase file folder in vscode( after these files/folders are pulled from git), it doesn't changes on file system.

**Comment by bpasero** [comment]
@kushalmahajan to clarify what you do:
* you open a folder with VSCode
* you click on a folder in the file explorer in VSCode
* you hit F2 to rename the casing of that folder
* the operation succeeds and you see your name in the file explorer in VSCode
* you look at the Windows explorer and the name of the folder has not changed

Is that accurate?

**Comment by kushalmahajan** [comment]
Yes.

All these steps hold true only if the folder/files are already been tracked by git with uppercase.

**Comment by bpasero** [comment]
@kushalmahajan any clue why using git would play a role here, it is not clear to me

@joaomoreno fyi

**Comment by kushalmahajan** [comment]
A lot has been done to resolve this afterwards.

I think it's git naming coming into play. The order of events is not quite fathamable but this should do.

`git mv old-file new-file` 

Doing this I also noticed that path on disk and git differs giving me some errors like not a git path.
Once I renamed from there for folders and files with the path git had previously it should be in sync with repo and disk for naming.

Closing this therefore for now
